src/test/java/com/dsrm/dsrmbackend/ReservationControllerTests.java

```diff
@@ -67,11 +69,19 @@ void addValidReservation() throws Exception {
         body.put("openingTime","2023-02-21 12:20:00");
         body.put("closingTime","2023-02-22 13:20:00");
         body.put("user",2);
-        this.mockMvc.perform(post("/reservations")
+        MvcResult result = this.mockMvc.perform(post("/reservations")
                         .contentType(MediaType.APPLICATION_JSON)
                         .content(objectMapper.writeValueAsString((body))))
                         .andExpect(status().isCreated())
-                        .andExpect(header().string("Location", "http://localhost/reservations/3"));
+                .andReturn();
+        String location = JsonPath.read(result.getResponse().getHeader("Location"), "$");
```

Comment:
We don't want to test GEt endpoint here. I would prefer if you would retrieve this data from repository. All changes in getter will break this test. It applies also to all tests below.